Backend/Backend/Controllers/AdminPanelController.cs

```diff
@@ -10,9 +12,34 @@ namespace Backend.Controllers;
 public class AdminPanelController : ControllerBase
 {
     private readonly AdminPanelService _adminService;
-    public AdminPanelController(AdminPanelService service) {
+    private readonly IAuthService _authService;
+    public AdminPanelController(AdminPanelService service, IAuthService authService) {
         _adminService = service;
+        _authService = authService;
     }
-    
-    
+
+    [HttpGet("users")]
+    public async Task<IActionResult> GetAllUsers() {
+        try {
+            return Ok(await _adminService.GetAllUsers());
+        }
+        catch (Exception e) {
+            return BadRequest(e.Message);
+        }
+    }
+
+    [HttpPost("ban/{userId}")]
+    public async Task<IActionResult> BanUser(Guid userId) {
+        try {
+            User? admin = await _authService.IdentifyUserAsync(HttpContext);
+            if (admin is null)
+                throw new InvalidOperationException("Admin user Id cannot be null");
```

Comment:
Error message is a bit off. This error happens when you are not logged in.

Reply:
This should never happen since the route is protected by a middleware. We need error message to be debuggable